### ISSUE TITLE
Tracking issue for version 0.4.0

### ISSUE THREAD

**Issue by copygirl** [copygirl/CarryCapacity#2]
Version 0.4.0 will introduce the ability to pick up blocks into a players' hands on as well as their back. Here's a couple of things that still have to be finished before a new version can be released:

### When carrying an item in hands
- [x] Render what's being carried when in first person. (7bf73021bcd6c0e53cc78ae94407e7c4872cba14)
- [x] Drop whatever you're carrying when getting damaged. (39f49acd45755cdba7e644735dd25047fd4c6a93)
- [x] Replace and lock hotbar slot so no items can be inserted.  
  (As well as off-hand slot now that it has been added.) (3ed1dd2c9160fe72e6a6249298faf5c7752d3996)
- [x] Prevent switching to a different hotbar slot. (See [VS issue #211](https://github.com/anegostudios/VintageStory-Issues/issues/211).) (7bf2cb9342ea8868742283586749d6c664c98c0e)
- [x] Prevent attacking / interacting when carrying something in-hand. (See [VS issue #208](https://github.com/anegostudios/VintageStory-Issues/issues/208).)
  Work-in-progress implemented in 525566bc0d90461943859652675669d3fe3cd1a3, still waiting for proper events.  
  Properly implemented in 1.9.4 with ed2228f52127aa1394a4136b2009b451f3947fd1.
- [x] Prevent sprinting when carrying something in-hand. (Mentioned in [VS issue #218](https://github.com/anegostudios/VintageStory-Issues/issues/218#issuecomment-430582503).) ( ed2228f52127aa1394a4136b2009b451f3947fd1)
  

### Other features / fixes
- [x] Test the old-version-attribute-updating with real worlds that used CarryCapacity.
  (Found some issues, fixed with d7dd6ac8d9bd56e13feb61e0ac85612b79f0cb21 and 9208dea70152032e6de807673c4946b14a1055af)
- [x] Rework block dropping code so they don't just drop in the exact same x,z vertical column the player died / got hit in, but instead place nearby if it makes sense. (20f4c694f691056583d674c5874b56f0812810f7)
- [x] Implement a `Shoulder` slot for carrying animals for [The Neolithic Mod](https://www.vintagestory.at/forums/topic/548-the-neolithic-mod/). Should include an animation where the player is holding the animal with left arm. Also prevent carrying in hands when something is on shoulder. (19c53906f0826e8d63f0cf566005dd7528ddf336)
- [ ] Display an icon in the center of the interaction circle depending on the current action (pick up, place down, or swap with back).
- [ ] Fix floating carried blocks at own player's feet when other players are out of render distance.
  (Appears to be a game engine bug. See [VS issue #216](https://github.com/anegostudios/VintageStory-Issues/issues/216).)
- [x] Fix animation not stopping upon dropping held blocks after taking damage. Does the server not have any control over the players' animations? (812204f93120b429c2183af1bd48977ca9b649f4)

**Comment by copygirl** [comment]
Updated the list: We also need to render what's being carried in-hand to the player now when they're in first person mode. Also to prevent hotbar switching I believe we do need a game patch as well.

**Comment by copygirl** [comment]
Tested out a private pre-release with HelDM and crew, found two issues, but they seem to be fixed. I belive the old-version-attribute-updating works now. Also added two new TODOs: Displaying an icon depending on current action in interaction circle and investigating the floating blocks at feet, which likely is some weirdness in the engine.

**Comment by copygirl** [comment]
While I was able to add the hotbar slot locking code, unfortunately my merge request hasn't made its way into 1.8 (at least up until 1.8.3 so far). May have to wait until 1.9 after all.

Additionally another issue came up with animations not being stopped if the server requests so?
